pkg/riverqueue/client.go

```diff
@@ -63,6 +63,12 @@ type Config struct {
 	RunMigrations bool `koanf:"runMigrations" json:"runMigrations" default:"false"`
 	// RiverConf is the river configuration
 	RiverConf river.Config `koanf:"riverConf" json:"riverConf"`
+	// EnableMetrics toggles otel metrics middleware
+	EnableMetrics bool `koanf:"enableMetrics" json:"enableMetrics" default:"false"`
+	// MetricsDurationUnit sets the duration unit for metrics
+	MetricsDurationUnit string `koanf:"metricsDurationUnit" json:"metricsDurationUnit" default:"ms"`
+	// EnableSemanticMetrics toggles semantic metrics
+	EnableSemanticMetrics bool `koanf:"enableSemanticMetrics" json:"enableSemanticMetrics" default:"true"`
 }
 
```

Comment:
Instead of putting these all on the top level config, it would make sense to make a metrics config an embed here, something like the following. This allows the metrics configuration to all be held together, which is important especially if more settings are needed in the future. 

```suggestion
	// Metrics configuration settings to enable and configure the metrics middleware
	Metrics MetricsConfig `koanf:"metrics" json:"metrics"`
}

// MetricsConfig allows the configuration of the metrics middleware
type MetricsConfig struct {
	// Enable toggles otel metrics middleware
	Enable bool `koanf:"enableMetrics" json:"enableMetrics" default:"false"`
	// DurationUnit sets the duration unit for metrics
	DurationUnit string `koanf:"metricsDurationUnit" json:"metricsDurationUnit" default:"ms"`
	// EnableSemanticMetrics toggles semantic metrics
	EnableSemanticMetrics bool `koanf:"enableSemanticMetrics" json:"enableSemanticMetrics" default:"true"`
}
```